### DISCUSSION TITLE
onRouteChange是在路由change时触发，有没有beforeChange时触发的

### DISCUSSION THREAD

**Discussion by hzl-stack** [umijs/umi#11828] [Q&A]
onRouteChange是在路由change时触发，有没有beforeChange时触发的

**Comment by fz6m** [comment]
如果只是想做路由跳转拦截和守卫，可以使用 `Prompt` ，参考 https://github.com/umijs/umi/discussions/10237#discussioncomment-4625915

还可以 patch `history.push` ，方式如下：

```ts
// src/app.ts

export const modifyClientRenderOpts = (context) => {
  const h = context.history
  const originPush = h.push
  const originReplace = h.replace
  h.push = (...args) => {
    // ...
    originPush.apply(h, args)
  }
  h.replace = (...args) => {
    // ...
    originReplace.apply(h, args)
  }
  return context
}
```

如果均为 hooks 调用，还可以使用 hooks 方式 patch ，可参考：[useReactRouterEnableConcurrentNavigation](https://github.com/SukkaW/foxact/blob/master/src/use-react-router-enable-concurrent-navigation/index.tsx)
